### ISSUE TITLE
SQLSTATE[HY000]: General error: 1364 Field 'disk_name' doesn't have a default value

### ISSUE THREAD

**Issue by aniket-magadum** [octobercms/october#5270]
- OctoberCMS Build: ### 468
- PHP Version: 7.2.19
- Database Engine: InnoDB
- Plugins Installed: RainLab.User

### Description:
I have created a component inside for my custom plugin and I am trying to attach one resume to the user model.

Plugin.php in the register method I have placed the following code.
```
$user->attachOne['resume'] = SystemFile::class;

```
This is the partial file

```htm
{% set user = __SELF__.user %}

<div class="profile-update">
    <h2>Additional Information</h2>
    {{ form_ajax(__SELF__~'::onUpdate',{files:true}) }}
    
    {% if user.resume %}
    <p>Uploaded Resume : <a href="{{ user.resume.getPath() }}">{{ user.resume.getFileName() }}</a> (You can
        upload a new resume below)</p>
    {% endif %}

    <div class="form-group">
        <label for="resume">Resume</label>
        <input type="file" class="form-control-file" name="resume" />
        <span class="text-danger" data-validate-for="resume"></span>
    </div>

    <button type="submit" class="btn btn-primary"> Save </button>
    {{ form_close()}}
</div>
```
I am uploading the file via AjaxHandler onUpdate(). Here is the code in the method.

```php
public function onUpdate()
{
     if (Input::hasFile('resume')) {
           $resume = Input::file('resume');
           $this->user->resume = $resume;
      }
      $this->user->save();
      return Redirect::refresh();
}
```

When the above code runs it throws an Exception as follows

```
"SQLSTATE[HY000]: General error: 1364 Field 'disk_name' doesn't have a default value (SQL: insert into `system_files` (`is_public`, `field`, `attachment_id`, `attachment_type`, `updated_at`, `created_at`) values (1, resume, 1, RainLab\User\Models\User, 2020-09-09 04:43:36, 2020-09-09 04:43:36))" on line 664 of C:\laragon\www\oc-recruitment\vendor\laravel\framework\src\Illuminate\Database\Connection.php
```

**Comment by damsfx** [comment]
I'm also having trouble with database attachment creation.  
 - OctoberCms build : 469
 - Php version : 7.3.11
 - Database Engine: InnoDB
 - No plugins

Either with : 
````
$stripe->logo = plugins_path() .'/acme/myplugin/assets/images/stripe.png';
$stripe->update();
````
or with : 
````
$stripe->logo()->create(['data' => realpath(__DIR__ . '../../assets/images/stripe.png')]);
````

**Comment by tobias-kuendig** [comment]
https://github.com/octobercms/library/commit/6cd6af6940ca15598ad1bab64e2db4a005d9d38f#diff-b062dfb9a3f1e4c504ad988c977f1b40 maybe?

**Comment by damsfx** [comment]
Reverting back to `protected $guarded = [];` seems to work in v**1.0.469** build.


But it's very strange ... This line has always been like this?
See [library/src/Database/Attach/File.php ](https://github.com/octobercms/library/blob/0fba71b3a3fc29191f9bffcd32e9c720199c90b3/src/Database/Attach/File.php#L33)  on 9 Jan 2014

**Comment by LukeTowers** [comment]
@bennothommo can you look into this?

**Comment by LukeTowers** [comment]
@damsfx @tobias-kuendig @4nik3t is this issue present in v1.1.0?

**Comment by damsfx** [comment]
Test this code (from command) with two fresh installs.

````php
$settings  = \Backend\Models\BrandSetting::instance();
$settings->logo = 'plugins/hounddd/toolbox/assets/images/logo.png';
$settings->save();
````
 - Php version : 7.2.11
 - Database : sqllite
 - No plugins

----

**October 1.1** : Work nice !
<details><summary>in composer.json</summary>
<p>

```json
"require": {
    "php": ">=7.2",
    "october/rain": "~1.1",
    "october/system": "~1.1",
    "october/backend": "~1.1",
    "october/cms": "~1.1",
    "laravel/framework": "~6.0",
    "wikimedia/composer-merge-plugin": "1.4.1",
    "offline/oc-bootstrapper": "^0.9.1"
},
```
</p>
</details>

----

**October 1.0** : **Error**.
<details><summary>in composer.json</summary>
<p>

```json
"require": {
    "php": ">=7.0.8",
    "ext-mbstring": "*",
    "ext-openssl": "*",
    "october/rain": "~1.0",
    "october/system": "~1.0",
    "october/backend": "~1.0",
    "october/cms": "~1.0",
    "laravel/framework": "~5.5.40",
    "wikimedia/composer-merge-plugin": "1.4.1",
    "offline/oc-bootstrapper": "^0.9.0"
},
```

</p>
</details>

Result this error
````batch
In Connection.php line 664:
  SQLSTATE[23000]: Integrity constraint violation: 19 NOT NULL constraint failed: system_files.disk_name (SQL: insert into "s
  ystem_files" ("is_public", "field", "attachment_id", "attachment_type", "updated_at", "created_at") values (1, logo, 1, Bac
  kend\Models\BrandSetting, 2020-09-10 23:10:12, 2020-09-10 23:10:12))

In PDOStatement.php line 107:
  SQLSTATE[23000]: Integrity constraint violation: 19 NOT NULL constraint failed: system_files.disk_name

In PDOStatement.php line 105:
  SQLSTATE[23000]: Integrity constraint violation: 19 NOT NULL constraint failed: system_files.disk_name
````
If I set `protected $guarded = []` in `library/src/Database/Attach/File.php` it works.

**Comment by bennothommo** [comment]
@LukeTowers Laravel have backported that security fix from L6 and L7 to L5.5 (the one which made guarded attributes that are not actual database columns no longer work) - https://github.com/laravel/framework/pull/33858

We'll have to add the changes from https://github.com/octobercms/library/commit/6cd6af6940ca15598ad1bab64e2db4a005d9d38f into the 1.0 branch too.

**Comment by bennothommo** [comment]
All,

If this is preventing you from doing work in the mean time, and you have used Composer, the simple fix is to change the `laravel/framework` dependency to `5.5.49` in your `composer.json` to downgrade L5.5, until such a time that we can release a new 1.0 build.

**Comment by LukeTowers** [comment]
Another simple fix would be to just upgrade to v1.1.0 via composer too :)

**Comment by bennothommo** [comment]
Fixed by https://github.com/octobercms/library/pull/521 - will be released with the next build of 1.0.
